src/components/transformation/create/PatchDraftButton.tsx

```diff
@@ -0,0 +1,138 @@
+import { Button } from '@mui/material';
+import { modifyDraftSpec } from 'api/draftSpecs';
+import {
+    useEditorStore_currentCatalog,
+    useEditorStore_persistedDraftId,
+    useEditorStore_queryResponse_mutate,
+    useEditorStore_setId,
+    useEditorStore_setPersistedDraftId,
+} from 'components/editor/Store/hooks';
+import { isEmpty } from 'lodash';
+import { useCallback } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useFormStateStore_setFormState } from 'stores/FormState/hooks';
+import { FormStatus } from 'stores/FormState/types';
+import {
+    useTransformationCreate_catalogName,
+    useTransformationCreate_migrations,
+    useTransformationCreate_setCatalogUpdating,
+    useTransformationCreate_setSelectedAttribute,
+    useTransformationCreate_transformConfigs,
+} from 'stores/TransformationCreate/hooks';
+import { Transform } from 'types';
+
+function PatchDraftButton() {
+    // Draft Editor Store
+    const currentCatalog = useEditorStore_currentCatalog();
+    const mutateDraftSpecs = useEditorStore_queryResponse_mutate();
+
+    const draftId = useEditorStore_persistedDraftId();
+    const setDraftId = useEditorStore_setId();
+    const setPersistedDraftId = useEditorStore_setPersistedDraftId();
+
+    // Form State Store
+    const setFormState = useFormStateStore_setFormState();
+
+    // Transformation Create Store
+    const catalogName = useTransformationCreate_catalogName();
+    const setSelectedAttribute = useTransformationCreate_setSelectedAttribute();
+
+    const setCatalogUpdating = useTransformationCreate_setCatalogUpdating();
+
+    const transformConfigs = useTransformationCreate_transformConfigs();
+    const migrations = useTransformationCreate_migrations();
+
+    const updateDerivationSpec = useCallback(async (): Promise<void> => {
```

Comment:
We don't need to change this now but before this goes live

This should use the non-nesting approach.